src/OmniSharp.Script/CachingScriptMetadataResolver.cs

```diff
@@ -29,35 +29,13 @@ public override int GetHashCode()
 
         public override PortableExecutableReference ResolveMissingAssembly(MetadataReference definition, AssemblyIdentity referenceIdentity)
         {
-            if (MissingReferenceCache.ContainsKey(referenceIdentity.Name))
-            {
-                return MissingReferenceCache[referenceIdentity.Name];
-            }
-
-            var result = _defaultReferenceResolver.ResolveMissingAssembly(definition, referenceIdentity);
-            if (result != null)
-            {
-                MissingReferenceCache[referenceIdentity.Name] = result;
-            }
-
-            return result;
+            return MissingReferenceCache.GetOrAdd(referenceIdentity.Name, _ => _defaultReferenceResolver.ResolveMissingAssembly(definition, referenceIdentity));
```

Comment:
I don't think it's the same thing as before. the old code would only cache non-nulls, whereas the new code would cache them.
In the second case it wouldn't cache empty collections too

Reply:
Yeah, that is true, But does it matter? If the `_defaultReferenceResolver` returns `null`, it would be `null` for any subsequent requests too?  So we save a call into the `_defaultReferenceResolver`. Unless that could change between request. If it actually is important we can replace this with `Tryadd` instead

Reply:
Or is it any other reason for this conditional caching? 😀